documentation.md

```diff
@@ -0,0 +1,50 @@
+# ocaml-tiff
+The ocaml-tiff library provides a pure OCaml implementation for reading and decoding TIFF (Tagged Image File Format) images. It is designed to parse TIFF structures, extract metadata, and reconstruct pixel data.
+
+
+[TIFF file format](https://en.wikipedia.org/wiki/TIFF)
+
+
+## Main file responsibilities in ocaml-tiff
+
+### tiff.ml 
```

Comment:
A bit like the tests, I think this kind of verbosity in the markdown file is hard to keep in sync (I see a #41 PR which might already make this outdated if it were to be merged). How about we shift these descriptions into comments in the head of the  `.ml` files?